File: spring-web-java/src/main/java/spring/web/java/domain/model/enums/OrderStatus.java
package spring.web.java.domain.model.enums;

import java.util.Arrays;

import com.fasterxml.jackson.annotation.JsonCreator;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum OrderStatus {
	ORDER("주문"),
	CONFIRM("확정"),
	CANCEL("취소");

	private final String description;

	@JsonCreator
	public static OrderStatus of(Object name) {
		return Arrays.stream(OrderStatus.values())
			.filter(orderStatus -> orderStatus.name().equals(name))
			.findFirst()
			.orElse(null);
	}
}

File: spring-web-kotlin/src/main/kotlin/spring/web/kotlin/domain/model/enums/OrderStatus.kt
package spring.web.kotlin.domain.model.enums

import com.fasterxml.jackson.annotation.JsonCreator

enum class OrderStatus(
    val description: String
) {
    ORDER("주문"),
    CONFIRM("확정"),
    CANCEL("취소");

    companion object {
        @JsonCreator
        fun of(name: String) = entries.find { it.name == name }
    }
}